### ISSUE TITLE
Catalog Child Forces do not Increment unit point cost without an application restart after every change

### ISSUE THREAD

**Issue by OftKilted** [BattleScribe/Release#11]
Note: This occurs and is repeatable on iOS and Android and a similar symptom exists on desktop

When you create a child force that can increment the number of models inside the unit the cost value for the unit stays stuck at the original value.

Example:
`Warhammer Age of Sigmar`
Add a `Order - Stormcast Eternals v12` parent force.
Add a `Order- Seraphon v19` child (allies) force.

Add a unit of `Saurus Guard`, The base cost is shown as 100 points. Increment the unit size by 1 increment 2x 5 Saurus Guard. Their points cost remains identical.

To work around the issue one must modify how the points are incremented for values to a 'for each increment of X' increase points by y.

To demonstrate:
to the list created above, add a `Order - Fyreslayers v10` force as a child force to the Stormcast Eternals force. (This catalog was created with a 'workaround mechanism' for the issue, but requires substantial work as this affects a multitude of units across all the catalogs.)

Select `Hearthguard Berzerkers` add a base selection. That gives a total of 100 points. Increment the unit size by 1, their points cost is now 200. 

This issue doesn't occur with two 'Parent' forces in the same roster. 
As an example:
Remove the  `Order- Seraphon v19` child (allies) force.
Add a  `Order- Seraphon v19` at the 'parent' level.

Select to add a `Saurus Guard` unit. The unit cost is correct, increment the unit size and the cost goes up accordingly.

**Comment by ghost** [comment]
Similar to the "next tick" problem with selections that have cost modifiers based on number of models in unit, where a change causes a corrective re-processing?

**Comment by OftKilted** [comment]
@capitaladot 
```
SSE for Main Unit (Saurus Guard) (cost 0) 
| - SSE - 5x Saurus Guard (100 points) (Min 1 selection in Parent)
```
When the number of blocks of `"5 Saurus Guard"` is increased to `2x 5 Saurus Guard ` units the points don't increment to 200 points if the force is a Child force.

If the force is not a child force, and is instead a parent force the points increment correctly.

**Comment by Jonskichov** [comment]
Fixed in desktop version 2.01.14.107

iOS/Android releases coming soon.

**Comment by Jonskichov** [comment]
Fixed in Android version 2.01.20.281

Should come through on the Play Store soon

**Comment by OftKilted** [comment]
@jonskichov We’re seeing odd points increment tracking after the update. (The points do appear.) But we have reports of “double counting”
I’ll do a follow up to let you know.

**Comment by Jonskichov** [comment]
Weird... Can you give an example?

**Comment by OftKilted** [comment]
We’re talking about it over in the AoS gitter

**Comment by OftKilted** [comment]
@jonskichov 
“Quickly looking at the issue myself it appears that the points are being doubled on the seraphon units, i.e a unit of 6 terrafon riders costs 240 and that registers correctly but is counting as 480 of allies”

**Comment by Jonskichov** [comment]
@oftkilted

"costs 240 and that registers correctly but is counting as 480 of allies"

I think we'll consider this a new issue since it sounds like the points of selections and roster total points are being calculated correctly.

This new issue seems to be to do with constraints. It might turn out to be a misconfigured constraint in the data file, but if not open a new issue and provide an example. Thanks! 😁

**Comment by Jonskichov** [comment]
Heads up, iOS fix was supposed to go up today but was held up by a slew of software updates (damnit WWDC). I need to check the build is working ok now that all that's done.

If it turns out we have a new issue, I will likely delay until a fix for that is in place too.

**Comment by Jonskichov** [comment]
Fixed in iOS version 2.01.18.117

It's submitted to the app store and will become available in a day or so after it's been reviewed.

I'll close this issue now rather than waiting for it to come out :D

**Comment by OftKilted** [comment]
:+1: The iOS release has hit! Thanks for the fix!
